feature/security/gnsi/acctz/tests/record_subscribe_non_grpc/record_subscribe_non_grpc_test.go

```diff
@@ -0,0 +1,677 @@
+		// straight away so this is easy button to trigger a shell record
+		shellUsername = "linuxadmin"
+		shellPassword = "NokiaSrl1!"
+	}
+
+	tcpConn, w, _ := dialSSH(t, shellUsername, shellPassword, target)
+	defer func() {
+		// give things a second to percolate then close the connection
+		time.Sleep(3 * time.Second)
+
+		err := tcpConn.Close()
+		if err != nil {
+			t.Logf("error closing tcp(ssh) connection, will ignore, error: %s", err)
+		}
+	}()
+
+	startTime := time.Now()
+
+	// this might not work for other vendors, so probably we can have a switch here and pass
+	// the writer to func per vendor if needed
+	_, err := w.Write([]byte(fmt.Sprintf("%s\n", shellCommand)))
+	if err != nil {
+		t.Fatalf("failed sending cli command, error: %s", err)
+	}
+
+	// remote from the perspective of the router
+	remoteAddr, err := net.ResolveTCPAddr("tcp", tcpConn.LocalAddr().String())
+	if err != nil {
+		t.Fatalf("failed resolving ssh remote addr, error: %s", err)
+	}
+	localAddr, err := net.ResolveTCPAddr("tcp", target)
+	if err != nil {
+		t.Fatalf("failed resolving ssh local addr, error: %s", err)
+	}
+
+	records = append(records, rpcRecord{
+		startTime:            startTime,
+		doneTime:             time.Now(),
+		cmdType:              acctz.CommandService_CMD_SERVICE_TYPE_SHELL,
+		localIp:              localAddr.IP.String(),
+		localPort:            uint32(localAddr.Port),
+		remoteIp:             remoteAddr.IP.String(),
+		remotePort:           uint32(remoteAddr.Port),
+		succeeded:            true,
+		expectedStatus:       acctz.SessionInfo_SESSION_STATUS_OPERATION,
+		expectedAuthenType:   acctz.AuthnDetail_AUTHN_TYPE_UNSPECIFIED,
+		expectedAuthenStatus: acctz.AuthnDetail_AUTHN_STATUS_UNSPECIFIED,
+		expectedAuthenCause:  "",
+		expectedIdentity:     shellUsername,
+	})
+
+	return records
+}
+
+func getDutAddr(t *testing.T, dut *ondatra.DUTDevice) string {
+	var serviceDUT interface {
+		Service(string) (*tpb.Service, error)
+	}
+
+	err := binding.DUTAs(dut.RawAPIs().BindingDUT(), &serviceDUT)
+	if err != nil {
+		t.Log("DUT does not support `Service` function, will attempt to use dut name field")
+
+		return dut.Name()
+	}
+
+	dutSSHService, err := serviceDUT.Service("ssh")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	return dutSSHService.GetOutsideIp()
+}
+
+func prettyPrint(i interface{}) string {
+	s, _ := json.MarshalIndent(i, "", "\t")
+	return string(s)
+}
+
+func TestAccountzRecordSubscribeNonGRPC(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+
+	setupUsers(t, dut)
+
+	// https://github.com/openconfig/featureprofiles/issues/2637
+	// basically, just waiting to see what the "best"/"preferred" way is to get the v4/v6 of the
+	// dut -- for now we use this hacky work around because ssh isn't exposed in introspection anyway
+	// so... we get what we can get.
+	addr := getDutAddr(t, dut)
+
+	// suppose ssh could be not 22 in some cases but don't think this is exposed by introspect
```

Comment:
Suppose

Reply:
and a period at the end of the sentence.

Reply:
Fixed comments